### ISSUE TITLE
Getting "unexpected HTTP response code 307 Temporary Redirect" on compile.

### ISSUE THREAD

**Issue by pineapplehunter** [tectonic-typesetting/tectonic#831]
Hi.
I could not download the packages when I tried to compile.

This was the output when compiling.
```sh
$ tectonic sample.tex 
note: connecting to https://archive.org/services/purl/net/pkgwpub/tectonic-default
error: unexpected HTTP response code 307 Temporary Redirect for URL https://archive.org/services/purl/net/pkgwpub/tectonic-default
```

I looked at the response and it seems that if you follow the link you can find the .tar file, so I think if you can follow the link this would be fixed.

**Comment by pineapplehunter** [comment]
I was looking at the code and I thing it has some thing to do with this redirect policy.
When I switched it to the default policy it seemed to work.
https://github.com/tectonic-typesetting/tectonic/blob/cde89ce6a26b63937432eec95c1e5fb8e1d5b390/crates/geturl/src/reqwest.rs#L46-L67

**Comment by pkgw** [comment]
I think the existing problem is an issue in reqwest, filed as seanmonstar/reqwest#1348.

I am working on setting up a dedicated redirection service, which should hopefully be live soon — the limiting factor has just been me getting the DNS straightened out.

I think it is actually better not to have two links in the binary. The service needs to be reliable, and adding multiple URL options just diffuses the responsibility for making sure that the service is working.

**Comment by pkgw** [comment]
Actually, no. The problem is in our custom redirection logic. There's at least one issue: Tectonic follows redirects as long as the new URL path has a final component that contains a `.`, with the idea that this recognizes redirects that look like filenames, but stops when we get to S3-type hashed storage. The new redirect stage from archive.org preserves the structure of our PURL URL, `/net/pkgwpub/tectonic-default`, which happens to not include a dot, so we immediately stop chasing redirects. This is definitely wrong behavior in this case.

It's also true that once we stop chasing redirects, we require that the most recent HTTP response code is either a 200 series or a 302, but something like a 307 is considered a failure. Based on how 307 is supposed to be essentially a more precisely-specified version of 302, we should almost surely accept it as a final result, although in this particular case it would have masked incorrect behavior.

I think I am going to change the redirect-chasing logic to also follow a redirect if the final path element of the new URL is the same as the final path element of the original URL. This seems like a good heuristic for ignoring S3-type redirects that ought to work for this particular case.

**Comment by pkgw** [comment]
Addressed in #832. I still hope to soon issue a new release that will include not only this fix, but an update of the builtin URL to a new service that will fix some of the other issues that have been associated with the archive.org PURL service.

**Comment by pkgw** [comment]
I've merged in #832, so binaries built from master (or, shortly, the latest continuous-deployment release) should be functional again.

**Comment by pkgw** [comment]
A new Tectonic release, 0.8.0, is now [processing in the CI systems][az], and should be finalized in a bit more than an hour.

[az]: https://dev.azure.com/tectonic-typesetting/tectonic/_build/results?buildId=799&view=results
